source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/AdaptiveCardRenderer.java

```diff
@@ -239,11 +247,18 @@ else if (alignment == ActionAlignment.Center.swigValue())
             viewGroup.addView(actionButtonsLayout);
         }
 
-        for (int i = 0; i < size && i < hostConfig.getActions().getMaxActions(); i++)
+        int i = 0;
+        int maxActions = (int) hostConfig.getActions().getMaxActions();
```

Comment:
if getMaxActions() return unsigned, would be better to have i unsigned?

Reply:
yeah. I should change this to long since that is what getMaxActions returns

Reply:
yeah. I should change this to long since that is what getMaxActions returns

---
In reply to: [175174429](https://github.com/Microsoft/AdaptiveCards/pull/1250#discussion_r175174429) [](ancestors = 175174429)

Reply:
Keeping i as an int but changing maxActions to long.

---
In reply to: [175220676](https://github.com/Microsoft/AdaptiveCards/pull/1250#discussion_r175220676) [](ancestors = 175220676,175174429)